ddl/partition.go

```diff
@@ -419,32 +419,39 @@ func buildTablePartitionInfo(ctx sessionctx.Context, s *ast.PartitionOptions, tb
 	var enable bool
 	switch s.Tp {
 	case model.PartitionTypeRange:
-		// When tidb_enable_table_partition is 'on' or 'auto'.
-		if s.Sub == nil {
-			// Partition by range expression is enabled by default.
-			if s.ColumnNames == nil {
-				enable = true
-			}
-			// Partition by range columns and just one column.
-			if len(s.ColumnNames) == 1 {
-				enable = true
-			}
+		// Partition by range expression is enabled by default.
+		if s.ColumnNames == nil {
+			enable = true
 		}
-	case model.PartitionTypeHash:
-		// Partition by hash is enabled by default.
-		// Note that linear hash is not enabled.
-		if !s.Linear && s.Sub == nil {
+		// Partition by range columns and just one column.
+		if len(s.ColumnNames) == 1 {
```

Comment:
This is the only modification compared with the original patch. It seems that range partition by more than 1 columns are not supported in 6.1. PTAL @mjonss

Reply:
That is correct, it was added to 6.5, see https://docs.pingcap.com/tidb/stable/basic-features#partitioning